src/silo/database.cpp

```diff
@@ -697,4 +759,10 @@ DataVersion Database::getDataVersion() const {
    return data_version_;
 }
 
+query_engine::QueryResult Database::executeQuery(const std::string& query) const {
+   const silo::query_engine::QueryEngine query_engine(*this);
+
+   return query_engine.executeQuery(query);
+}
```

Comment:
This kinda inverts the order of control. I find it more intuitive to have a query engine that holds a database (or the mutex in this case). That query engine lives somewhere as a singleton service that executes queries.

Why did you change it to this version? Maybe I can be convinced with some good arguments ;)

Reply:
As discussed: Let's leave as is, since the mutex only lives in silo_api and silo itself does not need to know about the mutex.